### PR TITLE
Add typescript parser, try indexing this repo itself.

### PR DESCRIPTION
**Pull request by icycodes**

### CODE REVIEW

**Comment by wsxiaoys** [comment]
#5

File: indexer/src/parser.ts [hunk 1]
```diff
@@ -28,6 +28,13 @@ const LanguageInfos: Array<LanguageInfo> = [
     maxLevel: 3,
     minLoc: 4,
   },
+  {
+    language: require("tree-sitter-typescript").typescript,
+    extensions: [".ts"],
```
Comment (wsxiaoys) [inline indexer/src/parser.ts:33]: does it supports tsx?

Comment (wsxiaoys) [reply indexer/src/parser.ts:33]: and js / jsx?

File: indexer/src/parser.ts [hunk 2]
```diff
@@ -28,6 +28,13 @@ const LanguageInfos: Array<LanguageInfo> = [
...
+    nodeTypes: ["function_declaration", "class_declaration"],
+    maxLevel: 3,
```
Comment (wsxiaoys) [inline indexer/src/parser.ts]: Have you verified the level settings? For now we only want to level function / class definitions.

File: indexer/src/parser.ts (hunk 1 above)
Comment (icycodes) [reply indexer/src/parser.ts:33]: No, need separate parser for each.

**Pull request merged**